### PR TITLE
improved update of gene panels

### LINKED ISSUE #866: gene panel upload - replace option
- [x] Add option to allow replacing gene panel on upload 
- [x] Multiple changes should be possible on same
- [x] UI should be clearly separated for what can be changed on edit, upload versus on creation of new panel
- [ ] Upload file should be able to have options `add`, `modify`, `replace`and `delete`

### PR DESCRIPTION
**Pull request by northwestwitch**
- prettier panels view
- Separated new panel creation from update of existing panel
- Introduced the add/replace genes option using a CSV file from panels view
- Fixed bug when adding/deleting/modifying a gene to panel multiple times
- Panel versioning is more flexible: custom version number chosen when applying pending changes is respected

fix #866
Anything missing???

### CODE REVIEW

**Comment by northwestwitch** [comment]
You can now test the code on stage: https://scout-stage.scilifelab.se/panels

File: scout/adapter/mongo/panel.py [hunk 1]
```diff
@@ -436,14 +440,26 @@ def apply_pending(self, panel_obj):
...
 
         new_panel['genes'] = new_genes
-        new_panel['version'] = panel_obj['version'] + 1
+        new_panel['version'] = float(version)
+
+        inserted_id = None
+        # if the same version of the panel should be updated
+        if new_panel['version'] == panel_obj['version']:
+            # replace panel_obj with new_panel
+            result = self.panel_collection.find_one_and_replace({'_id':panel_obj['_id']}, new_panel, return_document=pymongo.ReturnDocument.AFTER)
```
Comment (moonso) [inline scout/adapter/mongo/panel.py]:
```suggestion
            result = self.panel_collection.find_one_and_replace(
                {'_id':panel_obj['_id']}, 
                new_panel, 
                return_document=pymongo.ReturnDocument.AFTER
            )
```

File: scout/server/blueprints/panels/controllers.py [hunk 2]
```diff
@@ -27,47 +27,78 @@ def existing_gene(store, panel_obj, hgnc_id):
     return existing_genes.get(hgnc_id)
 
 
-def update_panel(store, panel_name, csv_lines):
-    """Update an existing gene panel with genes."""
+def update_panel(store, panel_name, csv_lines, option):
+    """Update an existing gene panel with genes.
```
Comment (moonso) [inline scout/server/blueprints/panels/controllers.py]: 👍

File: scout/server/blueprints/panels/controllers.py [hunk 3]
```diff
@@ -27,47 +27,78 @@ def existing_gene(store, panel_obj, hgnc_id):
...
         if gene_obj is None:
-            flash("gene not found: {} - {}".format(new_gene['hgnc_id'], new_gene['hgnc_symbol']),
-                  'danger')
+            flash("gene not found: {} - {}".format(new_gene['hgnc_id'], new_gene['hgnc_symbol']),'danger')
             continue
         if new_gene['hgnc_symbol'] and gene_obj['hgnc_symbol'] != new_gene['hgnc_symbol']:
-            flash("symbol mis-match: {} | {}".format(gene_obj['hgnc_symbol'],
-                  new_gene['hgnc_symbol']), 'warning')
-        action = 'edit' if gene_obj['hgnc_id'] in existing_genes else 'add'
+            flash("symbol mis-match: {} | {}".format(gene_obj['hgnc_symbol'],new_gene['hgnc_symbol']), 'warning')
```
Comment (moonso) [inline scout/server/blueprints/panels/controllers.py]:
```suggestion
            flash("symbol mis-match: {0} | {1}".format(
                gene_obj['hgnc_symbol'], new_gene['hgnc_symbol']), 'warning')
```

File: scout/server/blueprints/panels/views.py [hunk 4]
```diff
@@ -29,22 +29,28 @@ def panels():
...
-            flash("new gene panel added: {}!".format(panel_obj['panel_name']))
-        else:
-            panel_obj = controllers.update_panel(store, request.form['panel_name'], lines)
+            else:
+                flash("new gene panel added, {}!".format(new_panel_name))
+            return redirect(url_for('panels.panel', panel_id=new_panel_id))
+
+        else: # modify an existing panel
+            update_option = request.form['modify_option']
+            panel_obj= controllers.update_panel(store=store, panel_name=request.form['panel_name'], csv_lines=lines, option=update_option)
```
Comment (moonso) [inline scout/server/blueprints/panels/views.py]:
```suggestion
            panel_obj= controllers.update_panel(
                                   store=store, 
                                   panel_name=request.form['panel_name'], 
                                   csv_lines=lines, 
                                   option=update_option
             )
```

**Pull request merged**